src/aspire/utils/filters.py

```diff
@@ -73,6 +73,21 @@ def evaluate_grid(self, L, *args, **kwargs):
 
         return h
 
+    def dual(self):
+        return DualFilter(self)
```

Comment:
There's the question of whether we want to use `self` or `self.copy()`, so that subsequent tweaks to the current filter have no effect on the already-obtained dual. I don't know what the right answer is - if the rest of the code in this module is not bothering to copy filters then we're probably okay.

Reply:
Yes. I think it is safer to use `self.copy()`. But it depends on the implement of copy() method. When I do test using CTF filter, it returns AttributeError: 'CTFFilter' object has no attribute 'copy'.

Reply:
Thanks. I am curious why there are no default implementations of `__copy__` and `__deepcopy__`. I  will take a look today.

Reply:
Changed to use copy.deepcopy() to avoid recursively copying component objects in new commit 73ddfea.

Reply:
I'm not sure I understand the statement..`Changed to use copy.deepcopy() to avoid recursively copying component objects`. If avoiding recursive copying was your concern, then you shouldn't have used deepcopy.

I also didn't mean by this discussion that you should go ahead and do this - this is better discussed during our group meetings. I only meant it as a way to start thinking about this. Sorry if I gave you the wrong idea. Let's leave this open for now.